tests/__init__.py

```diff
@@ -110,8 +111,15 @@ def test_func(self):
                         msg=f"The actual value for .{key}() did not match the expected value.",
                     )
 
-        # Assert that the ingredients returned by the ingredient_groups() function
-        # are the same as the ingredients return by the ingredients() function.
+        # Special handling for ratings_count
+        if "ratings_count" in expect.keys():
```

Comment:
We probably want to relocate this earlier in the test class so that the `self.assertEqual` check _doesn't_ get evaluated for the field.

```py
    # Special handling for ratings_count
    expected_ratings_count = expect.pop("ratings_count")
    if expected_ratings_count:
        ...
```

...or something like that.

Reply:
Instead I explicitly excluded ratings_count from the `optional_tests` run so it can remain in the test listing

Reply:
Ok, thanks - I hadn't noticed previously that we run a subtest whether-or-not the key is in the `expect` dictionary.  I wonder if that's a bug?

Reply:
Ok, yep - I think that for the mandatory items, we do want to always run a subtest -- either checking the value, or raising an exception if no expectation exists.  However, for optional tests, we shouldn't create a subtest at all I don't think.

It's not directly relevant to this pull request though, so I'll open a separate pull request about that.